packages/typespec-ts/test/integration/azureClientGeneratorCoreAccess.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { assert } from "chai";
+import AccessClientFactory, {
+  AccessClient
+} from "./generated/azure/clientGeneratorCore/access/src/index.js";
+import { matrix } from "../util/matrix.js";
+
+describe("Access Client", () => {
+  let client: AccessClient;
+
+  interface UrlDetail {
+    type: string;
+    value: string;
+  }
+
+  const testedUrl: UrlDetail[] = [
+    {
+      type: "publicOperation",
+      value: "noDecoratorInPublic"
+    },
+    {
+      type: "publicOperation",
+      value: "publicDecoratorInPublic"
+    },
+    {
+      type: "internalOperation",
+      value: "noDecoratorInInternal"
+    },
+    {
+      type: "internalOperation",
+      value: "internalDecoratorInInternal"
+    },
+    {
+      type: "internalOperation",
+      value: "publicDecoratorInInternal"
+    },
+    {
+      type: "sharedModelInOperation",
+      value: "public"
+    },
+    { type: "sharedModelInOperation", value: "internal" }
+  ];
+
+  beforeEach(() => {
+    client = AccessClientFactory({
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  matrix([testedUrl], async (params: UrlDetail) => {
```

Comment:
In our practice the matrix is not recommanded because it would introduce casting like as any, so we can't verify type correctly.

Could you remove the matrix and sperate them one by one?

Reply:
I have removed the matrix and sperate them one by one